yarn-project/circuits.js/src/tests/factories.ts

```diff
@@ -260,46 +263,80 @@ export function makePrivateKernelInputs(seed = 1): PrivateKernelInputs {
 }
 
 export function makePublicCallStackItem(seed = 1): PublicCallStackItem {
-  return new PublicCallStackItem(
+  const callStackItem = new PublicCallStackItem(
     makeAztecAddress(seed),
-    new FunctionData(makeSelector(seed + 0x1), true, true),
+    // in the public kernel, function can't be a constructor or private
+    new FunctionData(makeSelector(seed + 0x1), false, false),
     makePublicCircuitPublicInputs(seed + 0x10),
   );
+  callStackItem.publicInputs.callContext.storageContractAddress = callStackItem.contractAddress;
+  return callStackItem;
 }
 
-export function makePublicCallData(seed = 1) {
-  return new PublicCallData(
+export async function makePublicCallData(seed = 1, wasm?: WasmWrapper) {
+  const publicCallData = new PublicCallData(
     makePublicCallStackItem(seed),
     range(PUBLIC_CALL_STACK_LENGTH, seed + 0x300).map(makePublicCallStackItem),
     makeProof(seed + 0x1000),
     fr(seed + 1),
     fr(seed + 2),
   );
+  // publicCallStack should be a hash of the preimages:
+  if (!wasm) {
```

Comment:
I think circuits wasm is a singleton instance so i think you can just call .get and carry on

Reply:
+1, let's simplify the interface and just `.get` the wasm whenever you need it